runtime/compiler/trj9/control/HookedByTheJit.cpp

```diff
@@ -7085,7 +7085,7 @@ int32_t setUpHooks(J9JavaVM * javaVM, J9JITConfig * jitConfig, TR_FrontEnd * vm)
             }
          }
 
-      if (!TR::Options::getCmdLineOptions()->getOption(TR_DisableJProfilerThread))
+      if (TR::Options::getCmdLineOptions()->getOption(TR_EnableJProfiling) && !TR::Options::getCmdLineOptions()->getOption(TR_DisableJProfilerThread))
```

Comment:
Unrelated to this PR, but why can JProfiling thread be enabled but JProfiling disabled? Shouldn't `!getOption(TR_DisableJProfilerThread)` always return the same as `getOption(TR_EnableJProfiling)`?

Reply:
Yes, that's another issue that needs to be corrected. Yi is aware of it.

Reply:
The reason the two options aren't set together is because JProfiling can currently be used in two ways: 1) use the JProfiling infrastructure in place of the current JITProfiling for profiling compilations (does not need the JProfilerThread per say) and 2) use JProfiling to instrument non-profiling compilations and allow for escalation. I believe this is the origin of the two flags.

Reply:
Yea, and also JProfilerThread is only needed if JProfiling is enabled.